From bc788a0f6ee9e6ca4736e690c04524d44598cd7f Mon Sep 17 00:00:00 2001
From: Kevin Falcone <falcone@bestpractical.com>
Date: Tue, 6 Sep 2011 17:02:08 -0400
Subject: [PATCH] Add two new callbacks

One to allow you to add new Attributes to the Class form before
submitting and one to allow you to have that field processed during
normal Class Updating.
---
 share/html/Admin/Articles/Classes/Modify.html |    3 +++
 1 files changed, 3 insertions(+), 0 deletions(-)

diff --git a/share/html/Admin/Articles/Classes/Modify.html b/share/html/Admin/Articles/Classes/Modify.html
index ac22fd9..892500e 100644
--- a/share/html/Admin/Articles/Classes/Modify.html
+++ b/share/html/Admin/Articles/Classes/Modify.html
@@ -90,6 +90,8 @@
 % }
 </ul>
 
+%$m->callback( CallbackName => 'BeforeSubmit', CustomFields => $cfs, ClassObj => $ClassObj );
+
 <& /Elements/Submit &>
 </form>
 
@@ -117,6 +119,7 @@ if ($Create) {
 if ($ClassObj->Id()) {
     $ARGS{HotList} ||= 0 if $Submitted;
     my @attribs= qw(Description Name HotList);
+    $m->callback( CallbackName => 'AttributeList', Attributes => \@attribs, ARGSRef => \%ARGS );
     push @results, UpdateRecordObject( AttributesRef => \@attribs,
                                        Object => $ClassObj,
                                        ARGSRef => \%ARGS);
-- 
1.7.4.1

